pydis_site/apps/api/models/bot/filters.py

```diff
@@ -0,0 +1,259 @@
+from .infraction import Infraction
+
+
+class FilterListType(models.IntegerChoices):
+    """Choice between allow or deny for a list type."""
+
+    ALLOW = 1
+    DENY = 0
+
+
+class FilterList(ModelTimestampMixin, ModelReprMixin, models.Model):
+    """Represent a list in its allow or deny form."""
+
+    name = models.CharField(max_length=50, help_text="The unique name of this list.")
+    list_type = models.IntegerField(
+        choices=FilterListType.choices,
+        help_text="Whether this list is an allowlist or denylist"
+    )
+    dm_content = models.CharField(
+        max_length=1000,
+        null=False,
+        blank=True,
+        help_text="The DM to send to a user triggering this filter."
+    )
+    dm_embed = models.CharField(
+        max_length=2000,
+        help_text="The content of the DM embed",
+        null=False,
+        blank=True
+    )
+    infraction_type = models.CharField(
+        choices=[
+            (choices[0].upper(), choices[1])
+            for choices in [("NONE", "None"), *Infraction.TYPE_CHOICES]
+        ],
+        max_length=10,
+        null=False,
+        help_text="The infraction to apply to this user."
+    )
+    infraction_reason = models.CharField(
+        max_length=1000,
+        help_text="The reason to give for the infraction.",
+        blank=True,
+        null=False
+    )
+    infraction_duration = models.DurationField(
+        null=False,
+        help_text="The duration of the infraction. 0 for permanent."
+    )
+    infraction_channel = models.BigIntegerField(
+        validators=(
+            MinValueValidator(
+                limit_value=0,
+                message="Channel IDs cannot be negative."
+            ),
+        ),
+        help_text="Channel in which to send the infraction.",
+        null=False
+    )
+    guild_pings = ArrayField(
+        models.CharField(max_length=100),
+        help_text="Who to ping when this filter triggers.",
+        null=False
+    )
+    filter_dm = models.BooleanField(help_text="Whether DMs should be filtered.", null=False)
+    dm_pings = ArrayField(
+        models.CharField(max_length=100),
+        help_text="Who to ping when this filter triggers on a DM.",
+        null=False
+    )
+    remove_context = models.BooleanField(
+        help_text=(
+            "Whether this filter should remove the context (such as a message) "
+            "triggering it."
+        ),
+        null=False
+    )
+    bypass_roles = ArrayField(
+        models.CharField(max_length=100),
+        help_text="Roles and users who can bypass this filter.",
+        null=False
+    )
+    enabled = models.BooleanField(
+        help_text="Whether this filter is currently enabled.",
+        null=False
+    )
+    send_alert = models.BooleanField(
+        help_text="Whether an alert should be sent.",
+    )
+    # Where a filter should apply.
+    enabled_channels = ArrayField(
+        models.CharField(max_length=100),
```

Comment:
Channels and categories can be specified by their names as well.

Reply:
Names can change. What about a filter on one of the off-topic channels?

Reply:
Then for those you can use the channel ID. Name support was added to match the current filters which don't trigger in categories called "CODE JAM". I expect that we will mostly use IDS, but it might come in handy in additional scenarios in the future.